src/lib.rs

```diff
@@ -126,6 +192,24 @@ fn get_identifiers(idents: &Option<CustomIdentifiers>) -> Identifiers {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Tsify)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+enum KeyStretchingFunctionConfig {
+    #[serde(rename = "recommended")]
+    Recommended,
+    #[serde(rename = "memory-constrained")]
+    MemoryConstrained,
+    #[serde(rename = "argon2id-custom")]
+    Custom {
+        #[serde(rename = "iterations")]
+        iterations: u32,
+        #[serde(rename = "memory")]
+        memory: u32,
```

Comment:
It would be great if memory could also be given by `KBs`, `MBs`, and `GBs`, for convenience. I could see that being a lot more useful to users rather than straight bytes. We could create an easy conversion method to convert them 

Something like:
```
enum MemoryType {
    Bytes,
    KBs,
    MBs,
    GBs,
}

struct MemoryUsage {
    memory_type: MemoryType,
    value: u32,
}

fn memory_usage_to_bytes(usage: MemoryUsage) -> u32 {
    return match usage.memory_type {
        MemoryType::Bytes => usage.value,
        MemoryType::KBs => usage.value * 1000,
        MemoryType::MBs => usage.value * 1000000,
        MemoryType::GBs => usage.value * 1e+9,
    };
}
```

Reply:
How would this look like in JavaScript when you use the API?

Reply:
I was thinking memory would just be an object instead of a `u32`, with `memory_type` being `Optional` (defaulted to `Bytes` if not specified)?

Something like:
```
    "argon2id-custom": {
      memory: {
            value: 65536   // defaulted to "bytes" since not specified?
       },
      iterations: 1,
      parallelism: 4,
    }
```

or 
```
    "argon2id-custom": {
      memory: {
            memoryType: "mbs",
            value: 64
       },
      iterations: 1,
      parallelism: 4,
    }
```